server/athenian/api/controllers/goal_controller.py

```diff
@@ -168,4 +182,60 @@ async def parse_request_repositories(
 
 async def measure_goals(request: AthenianWebRequest, body: dict) -> web.Response:
     """Calculate the metrics for the goal tree."""
-    raise NotImplementedError()
+    goals_request = AlignGoalsRequest.from_dict(body)
+    team = goals_request.team
+    team_rows, meta_ids, jira_config = await gather(
+        fetch_teams_recursively(
+            goals_request.account,
+            request.sdb,
+            select_entities=(Team.id, Team.name, Team.members, Team.parent_id),
+            # teamId 0 means to implicitly use the single root team
+            root_team_ids=None if team == 0 else [team],
+        ),
+        get_metadata_account_ids(goals_request.account, request.sdb, request.cache),
+        get_jira_installation_or_none(
+            goals_request.account, request.sdb, request.mdb, request.cache,
+        ),
+    )
+    team_tree = build_team_tree_from_rows(team_rows, None if team == 0 else team)
+    team_member_map = flatten_teams(team_rows)
+
+    team_ids = [row[Team.id.name] for row in team_rows]
+    team_goal_rows, prefixer = await gather(
+        fetch_team_goals(goals_request.account, team_ids, request.sdb),
+        Prefixer.load(meta_ids, request.mdb, request.cache),
+    )
+
+    goals_to_serve = []
+    # iter all team goal rows, grouped by goal, to build GoalToServe object for the goal
+    for _, group_team_goal_rows_iter in groupby(team_goal_rows, itemgetter(Goal.id.name)):
+        goal_team_goal_rows = list(group_team_goal_rows_iter)
+        goal_to_serve = GoalToServe(
+            goal_team_goal_rows,
+            team_tree,
+            team_member_map,
+            prefixer,
+            goals_request.only_with_targets,
+            jira_config,
+        )
+        goals_to_serve.append(goal_to_serve)
+
+    all_metric_values = await calculate_team_metrics(
+        [g.request for g in goals_to_serve],
+        account=goals_request.account,
+        meta_ids=meta_ids,
+        sdb=request.sdb,
+        mdb=request.mdb,
+        pdb=request.pdb,
+        rdb=request.rdb,
+        cache=request.cache,
+        slack=request.app["slack"],
+        unchecked_jira_config=jira_config,
+    )
+
+    goal_tree_generator = GoalTreeGenerator()
+    models = [
+        to_serve.build_goal_tree(all_metric_values, goal_tree_generator).to_dict()
```

Comment:
`to_dict()` should always happen inside `model_response()`. It is easier to tune serialization in the future this way.